server.py

```diff
@@ -156,26 +157,25 @@ def _estimate_with_deadline(self, product, deadline=600):
         it a queue to return a (result, exception-string) through.
         The process can time out, and the queue can be empty - both need to provide
         exceptions in addition to anything received via the queue."""
-        q = multiprocessing.Queue(1)
-        p = multiprocessing.Process(target=self._estimate_outside_process, args=(product, q))
+        q = ctx.Queue(1)
+        p = ctx.Process(target=self._estimate_outside_process, args=(product, q))
+        results = (None, "no results yet")
         try:
             p.start()
             self.logging.info(f"🍴 Forked {p.pid} to compute estimation")
-            p.join(deadline)
-            if p.is_alive():
+            try:
+                results = q.get(block=True, deadline)
```

Comment:
```suggestion
                results = q.get(block=True, timeout=deadline)
```

Test are failing because of this (you can't put non keywords args after a keyword arg)